builders/txDataBuilder.go

```diff
@@ -0,0 +1,184 @@
+package builders
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+)
+
+const dataSeparator = "@"
+
+// txDataBuilder can be used to easy construct a transaction's data field for a smart contract call
+// can also be used to construct a VmValueRequest instance ready to be used on a VM query
+type txDataBuilder struct {
+	address    string
+	function   string
+	callerAddr string
+	args       []string
+	log        logger.Logger
+	err        error
+}
+
+// NewTxDataBuilder creates a new transaction data builder
+func NewTxDataBuilder(log logger.Logger) *txDataBuilder {
+	builder := &txDataBuilder{
+		log: log,
```

Comment:
please remove the log param. this component should be simply enough so it shouldn't have any parameter on the constructor

Reply:
removed, unused